File: main/src/main/java/com/sedmelluq/discord/lavaplayer/source/soundcloud/SoundCloudFormatHandler.java
package com.sedmelluq.discord.lavaplayer.source.soundcloud;

import java.util.List;

public interface SoundCloudFormatHandler {
    SoundCloudTrackFormat chooseBestFormat(List<SoundCloudTrackFormat> formats);

    String buildFormatIdentifier(SoundCloudTrackFormat format);

    SoundCloudM3uInfo getM3uInfo(String identifier);

    String getMp3LookupUrl(String identifier);
}

File: main/src/main/java/com/sedmelluq/discord/lavaplayer/source/soundcloud/SoundCloudFormatHandler.kt
package com.sedmelluq.discord.lavaplayer.source.soundcloud

interface SoundCloudFormatHandler {
    fun chooseBestFormat(formats: List<SoundCloudTrackFormat>): SoundCloudTrackFormat

    fun buildFormatIdentifier(format: SoundCloudTrackFormat): String

    fun getM3uInfo(identifier: String): SoundCloudM3uInfo?

    fun getMp3LookupUrl(identifier: String): String?
}
